apps/linode-marketplace-gitea/roles/gitea/tasks/postgresql.yml

```
@@ -0,0 +1,52 @@
+---
+- name: Add PostgreSQL repository key
+  apt_key:
+    url: https://www.postgresql.org/media/keys/ACCC4CF8.asc
+    state: present
+
+- name: Add PostgreSQL repository
+  apt_repository:
+    repo: deb http://apt.postgresql.org/pub/repos/apt/ {{ ansible_distribution_release }}-pgdg main
+    state: present
+    filename: pgdg
+
+- name: Install PostgreSQL 16 and related packages
+  apt:
+    name:
+      - postgresql-16
+      - postgresql-contrib-16
+    state: present
+    update_cache: true
+
+- name: Enable and start PostgreSQL service
+  systemd:
+    name: postgresql
+    state: started
+    enabled: true
+
+- name: Create PostgreSQL user and database
+  become: true
+  become_user: postgres
+  community.postgresql.postgresql_user:
+    name: gitea
+    password: "{{ postgresql_password }}"
+    role_attr_flags: NOSUPERUSER,NOCREATEDB,NOCREATEROLE
+    state: present
+
+- name: Create PostgreSQL database with proper owner
+  become: true
+  become_user: postgres
+  community.postgresql.postgresql_db:
+    name: giteadb
+    owner: gitea
+    encoding: UTF8
+    template: template0
+    state: present
```

```suggestion
- name: Create PostgreSQL user and database
  community.postgresql.postgresql_user:
    name: gitea
    password: "{{ postgresql_password }}"
    role_attr_flags: NOSUPERUSER,NOCREATEDB,NOCREATEROLE
    state: present
  become: true
  become_user: postgres

- name: Create PostgreSQL database with proper owner
  community.postgresql.postgresql_db:
    name: giteadb
    owner: gitea
    encoding: UTF8
    template: template0
    state: present
  become: true
  become_user: postgres
```
feel free to ignore this one, looks like a majority of the playbooks put the become user at the bottom, nothing functionally thats wrong here, just looks weird to me with them on top.